third_party/libunwindstack/PeCoffInterface.cpp

```diff
@@ -252,6 +261,113 @@ bool PeCoffInterface<AddressType>::ParseOptionalHeader(uint64_t offset) {
     }
   }
 
+  coff_memory_.set_cur_offset(end_offset);
+
+  return true;
+}
+
+template <typename AddressType>
+bool PeCoffInterface<AddressType>::ParseSectionHeaders(uint64_t offset) {
+  uint16_t num_sections = coff_header_.nsects;
+
+  coff_memory_.set_cur_offset(offset);
+
+  for (uint16_t idx = 0; idx < num_sections; ++idx) {
+    SectionHeader section_header;
+
+    // Section names in the header are always exactly 8 bytes. Longer names have to be looked
+    // up in the string table.
+    if (!coff_memory_.GetFully(static_cast<void*>(&section_header.name[0]), 8)) {
+      last_error_.code = ERROR_MEMORY_INVALID;
+      last_error_.address = coff_memory_.cur_offset();
+      Log::Error("Parsing section header failed: %s", GetErrorCodeString(last_error_.code));
+      return false;
+    }
+
+    if (!coff_memory_.Get32(&section_header.vmsize) ||
+        !coff_memory_.Get32(&section_header.vmaddr) || !coff_memory_.Get32(&section_header.size) ||
+        !coff_memory_.Get32(&section_header.offset) ||
+        !coff_memory_.Get32(&section_header.reloff) ||
+        !coff_memory_.Get32(&section_header.lineoff) || !coff_memory_.Get16(&section_header.nrel) ||
+        !coff_memory_.Get16(&section_header.nline) || !coff_memory_.Get32(&section_header.flags)) {
+      last_error_.code = ERROR_MEMORY_INVALID;
+      last_error_.address = coff_memory_.cur_offset();
+      Log::Error("Parsing section header failed: %s", GetErrorCodeString(last_error_.code));
+      return false;
+    }
+    parsed_section_headers_.emplace_back(section_header);
+  }
+  return true;
+}
+
+template <typename AddressType>
+bool PeCoffInterface<AddressType>::GetSectionName(const std::string& parsed_section_name_string,
+                                                  std::string* result) {
+  uint64_t offset = 0;
+  if (!android::base::ParseUint<uint64_t>(parsed_section_name_string.c_str(), &offset)) {
+    Log::Error("Failed to parse section name as integer: %s", parsed_section_name_string.c_str());
+    last_error_.code = ERROR_INVALID_COFF;
+    return false;
+  }
+
+  // The symbols come first and every one of them has a size of 18 bytes and we need to add
+  // this to the offset to get to the strings for the section names.
+  uint64_t file_offset = coff_header_.symoff + (18 * coff_header_.nsyms) + offset;
```

Comment:
Yeah, fully agree. I think it actually helps quite a bit here (and in other places where `const` is now added after your comment), because there are also many place where offsets are updated throughout methods. Adding `const` where possible really makes it easier to distinguish between these cases.